src/coinjoin/coinjoin.h

```diff
@@ -322,7 +322,7 @@ class CCoinJoinBroadcastTx
     bool Sign();
     [[nodiscard]] bool CheckSignature(const CBLSPublicKey& blsPubKey) const;
 
-    void SetConfirmedHeight(int nConfirmedHeightIn) { nConfirmedHeight = nConfirmedHeightIn; }
+    void SetConfirmedHeight(std::optional<int> nConfirmedHeightIn) { nConfirmedHeight = nConfirmedHeightIn; }
```

Comment:
```suggestion
    void SetConfirmedHeight(std::optional<int> nConfirmedHeightIn) { assert(nConfirmedHeightIn > 0); nConfirmedHeight = nConfirmedHeightIn; }
```

Reply:
- Why?

Reply:
cause it can't be negative now
(modified the suggestion)